diff --git a/configure.ac b/configure.ac
index fb14215..8e0cb4a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -119,7 +119,7 @@ AC_ARG_WITH(gtk, [
 AM_CONDITIONAL(WITH_GTK, test "$with_gtk" != "no")
 
 if test "x$with_gtk" != "xno"; then
-	PKG_CHECK_MODULES(GTK, gtk+-3.0 >= $GTK_REQ gtk+-x11-3.0 >= $GTK_REQ)
+	PKG_CHECK_MODULES(GTK, gtk+-3.0 >= $GTK_REQ)
 	AC_SUBST(GTK_CFLAGS)
 	AC_SUBST(GTK_LIBS)
 fi
diff --git a/ui/frob-prompt.c b/ui/frob-prompt.c
index 20462f6..86a64f8 100644
--- a/ui/frob-prompt.c
+++ b/ui/frob-prompt.c
@@ -25,8 +25,9 @@
 #include "gcr/gcr-base.h"
 
 #include <gtk/gtk.h>
+#if 0
 #include <gdk/gdkx.h>
-
+#endif 
 #include <err.h>
 
 static const gchar *file_name = NULL;
@@ -78,9 +79,11 @@ prompt_perform (GtkWidget *parent)
 		errx (1, "couldn't create prompt: %s", error->message);
 
 	if (parent) {
+#if 0
 		caller_id = g_strdup_printf ("%lu", (gulong)GDK_WINDOW_XID (gtk_widget_get_window (parent)));
 		gcr_prompt_set_caller_window (GCR_PROMPT (prompt), caller_id);
 		g_free (caller_id);
+#endif 
 	}
 
 	loop = g_main_loop_new (NULL, FALSE);
diff --git a/ui/frob-system-prompt.c b/ui/frob-system-prompt.c
index e08114f..9912009 100644
--- a/ui/frob-system-prompt.c
+++ b/ui/frob-system-prompt.c
@@ -26,8 +26,9 @@
 #include "egg/egg-testing.h"
 
 #include <gtk/gtk.h>
+#if 0
 #include <gdk/gdkx.h>
-
+#endif
 #include <unistd.h>
 #include <string.h>
 #include <errno.h>
@@ -53,11 +54,11 @@ on_prompt_clicked (GtkToolButton *button,
 	gcr_prompt_set_title (GCR_PROMPT (prompt), "This is the title");
 	gcr_prompt_set_message (GCR_PROMPT (prompt), "This is the message");
 	gcr_prompt_set_description (GCR_PROMPT (prompt), "This is the description");
-
+#if 0
 	caller_id = g_strdup_printf ("%lu", (gulong)GDK_WINDOW_XID (gtk_widget_get_window (parent)));
 	gcr_prompt_set_caller_window (GCR_PROMPT (prompt), caller_id);
 	g_free (caller_id);
-
+#endif
 	password = gcr_prompt_password_run (GCR_PROMPT (prompt), NULL, &error);
 	if (error != NULL) {
 		g_warning ("couldn't prompt for password: %s", error->message);
diff --git a/ui/gcr-prompt-dialog.c b/ui/gcr-prompt-dialog.c
index 606620e..6430f44 100644
--- a/ui/gcr-prompt-dialog.c
+++ b/ui/gcr-prompt-dialog.c
@@ -27,7 +27,9 @@
 #include "gcr-secure-entry-buffer.h"
 
 #include <gtk/gtk.h>
+#if 0
 #include <gdk/gdkx.h>
+#endif 
 #include <glib/gi18n.h>
 
 /**
@@ -147,13 +149,16 @@ update_transient_for (GcrPromptDialog *self)
 		g_warning ("couldn't parse caller-window property: %s", self->pv->caller_window);
 		return;
 	}
-
+#if 0
 	display = gtk_widget_get_display (GTK_WIDGET (self));
 	transient_for = gdk_x11_window_foreign_new_for_display (display, (Window)handle);
 	if (transient_for == NULL) {
 		g_warning ("caller-window property doesn't represent a window on current display: %s",
 		           self->pv->caller_window);
-	} else {
+	} else 
+#endif
+        {
+
 		gdk_window_set_transient_for (window, transient_for);
 		g_object_unref (transient_for);
 	}
diff --git a/ui/gcr-prompter-tool.c b/ui/gcr-prompter-tool.c
index cd1cc56..8b49423 100644
--- a/ui/gcr-prompter-tool.c
+++ b/ui/gcr-prompter-tool.c
@@ -27,7 +27,9 @@
 
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
+#if 0
 #include <gdk/gdkx.h>
+#endif 
 #include <pango/pango.h>
 
 #include <locale.h>
